ui/src/app/views/cohort-edit/component.ts

```diff
@@ -3,52 +3,56 @@ import {Router, ActivatedRoute} from '@angular/router';
 
 import {Cohort} from 'generated';
 import {CohortsService} from 'generated';
-import {User} from 'app/models/user';
-import {UserService} from 'app/services/user.service';
-import {CohortEditService} from 'app/services/cohort-edit.service';
+import {WorkspaceComponent} from 'app/views/workspace/component';
 
 @Component({
   styleUrls: ['./component.css'],
   templateUrl: './component.html',
 })
 export class CohortEditComponent implements OnInit {
-  user: User;
   cohort: Cohort = {id: '', name: '', description: '', criteria: '', type: ''};
   cohortId: string;
   adding = false;
   constructor(
       private router: Router,
       private route: ActivatedRoute,
-      private userService: UserService,
       private cohortsService: CohortsService,
-      private CohortEditService: CohortEditService
   ) {}
 
   ngOnInit(): void {
-    this.userService.getLoggedInUser()
-        .then(user => this.user = user);
     this.cohortId = this.route.snapshot.url[4].path;
     if (this.route.snapshot.url[5] === undefined) {
       this.adding = true;
     } else {
-      this.CohortEditService.get(this.cohortId).then(cohort => {
-        this.cohort = cohort;
-      }).catch(error => {
-        alert('All of Us Researcher Workbench does not currently \
-support editing cohorts from the API.');
-        this.router.navigate(['../../..'], {relativeTo : this.route});
-      });
+      this.cohortsService
+          .getCohort(
+              WorkspaceComponent.DEFAULT_WORKSPACE_NS,
+              WorkspaceComponent.DEFAULT_WORKSPACE_ID,
+              this.cohortId)
+          .retry(2)
+          .subscribe(cohort => { this.cohort = cohort; });
     }
   }
 
   saveCohort(): void {
-    this.CohortEditService.edit(this.cohort.id, this.cohort).then(
-      cohorts => this.router.navigate(['../../..'], {relativeTo : this.route}));
+    this.cohortsService
+        .updateCohort(
+            WorkspaceComponent.DEFAULT_WORKSPACE_NS,
+            WorkspaceComponent.DEFAULT_WORKSPACE_ID,
+            this.cohort.id,
+            this.cohort)
+        .retry(2)
+        .subscribe(cohorts => this.router.navigate(['../../..'], {relativeTo : this.route}));
   }
 
   addCohort(): void {
-    this.CohortEditService.add(this.cohort).then(
-      cohorts => this.router.navigate(['../..'], {relativeTo : this.route}));
+    this.cohortsService
+        .createCohort(
+            WorkspaceComponent.DEFAULT_WORKSPACE_NS,
+            WorkspaceComponent.DEFAULT_WORKSPACE_ID,
+            this.cohort)
```

Comment:
I think we should revisit our thoughts on ID, because the ID following the name, but being immutable, while the name is mutable, means they can and probably will easily become decoupled.